Tests/MapboxMapsTests/Snapshot/SnapshotterTests.swift

```diff
@@ -5,24 +5,111 @@ final class SnapshotterTests: XCTestCase {
 
     var mapboxObservableProviderStub: Stub<ObservableProtocol, MapboxObservableProtocol>!
     var snapshotter: Snapshotter!
+    var mockMapSnapshotter: MockMapSnapshotter!
 
     override func setUp() {
         super.setUp()
         let options = MapSnapshotOptions(
             size: CGSize(width: 100, height: 100),
             pixelRatio: .random(in: 1...3))
         mapboxObservableProviderStub = Stub(defaultReturnValue: MockMapboxObservable())
+        mockMapSnapshotter = MockMapSnapshotter()
         snapshotter = Snapshotter(
             options: options,
-            mapboxObservableProvider: mapboxObservableProviderStub.call(with:))
+            mapboxObservableProvider: mapboxObservableProviderStub.call(with:),
+            mapSnapshotter: mockMapSnapshotter)
     }
 
     override func tearDown() {
         snapshotter = nil
         mapboxObservableProviderStub = nil
+        mockMapSnapshotter = nil
         super.tearDown()
     }
 
+    // Test snapshot start invokes mockMapSnapshotter startStub
+    func testSnapshotterStartInvocation() {
+        // asserts that the mock and snapshotter reach start
+        snapshotter.start(overlayHandler: nil) { (result) in
+            XCTAssertEqual(self.mockMapSnapshotter.startStub.invocations.count, 1)
+        }
+    }
+
+    // asserting that the snapshotter image and mock snapshotter image are the same
+    func testSnapshotterImage() {
+        let snapshotView = UIImageView()
+        var snapshotting: Bool?
+        snapshotter.start(overlayHandler: nil) { ( result ) in
+            switch result {
+            case .success(let image):
+                snapshotView.image = image
+                XCTAssertEqual(image, self.mockMapSnapshotter.image)
+            case .failure(let error):
+                print("Error generating snapshot: \(error)")
+            }
+            snapshotting = false
+        }
+    }
+
+    // Test snapshot cancellation invokes mockMapSnapshotter cancelStub
+    func testSnapshotterCancel() {
+        // given snapshot is cancelled
+        snapshotter.cancel()
+        // mockMapSnapshotter cancel snapshotter stub should equal that of snapshotter
+        XCTAssertEqual(mockMapSnapshotter.cancelSnapshotterStub.invocations.count, 1)
+    }
+
+    // TestSnapshot size is the same size as the mock snapshot
+    func testSnapshotterSize() {
+        let size = CGSize(width: 200, height: 200)
+
+        snapshotter.mapSnapshotter.setSizeFor(.init(size))
+
+        XCTAssertEqual(snapshotter.snapshotSize, CGSize(mockMapSnapshotter.getSize()))
+    }
+
+    // TestSnapshot size shares same tile mode as the mock snapshot
+    func testSnapshotterTileMode() {
+
+        snapshotter.mapSnapshotter.setTileModeForSet(true)
+
+        XCTAssertEqual(mockMapSnapshotter.isInTileModeStub.invocations.count, 1)
```

Comment:
This assertion fails for me. Is it testing something different from the one below (which is passing)?

Reply:
I actually fixed that one, just haven't pushed up my changes